lib/interfaces/bdd.js

```diff
@@ -22,12 +30,27 @@ const addStep = (step, fn) => {
 
 const parameterTypeRegistry = new ParameterTypeRegistry();
 
-const matchStep = (step) => {
+const _findClosestPath = (from, tos) => {
+  let _shortest = Infinity;
+  let _closest = tos[0];
+  for (const each of tos) {
+    let to = each.line;
+    to = to.slice(to.indexOf(path.sep) - 1, to.lastIndexOf(path.sep) + 1);
+    const _distance = (path.relative(from, to).split(path.sep).length - 1);
+    if (_distance < _shortest) {
+      _shortest = _distance;
+      _closest = each;
+    }
+  }
+  return _closest;
+};
+
+const matchStep = (step, featureName) => {
   for (const stepName in steps) {
     if (stepName.indexOf('/') === 0) {
       const res = step.match(new RegExp(stepName.slice(1, -1)));
       if (res) {
-        const fn = steps[stepName];
+        const fn = (featureName && Object.keys(matchingSteps).includes(stepName)) ? _findClosestPath(featureName, matchingSteps[stepName]) : steps[stepName];
```

Comment:
sorry, this code is so complicated that I can't understand it.
Please expand this construct so it would be possible to understand what happens here